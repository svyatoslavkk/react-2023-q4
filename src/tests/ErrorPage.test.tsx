import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ErrorPage from '../pages/404';

vi.mock('next/router', () => vi.importActual('next-router-mock'));

describe('Tests for the 404 Page component', () => {
  it('404 page is displayed when navigating to an invalid route.', async () => {
    const page = render(<ErrorPage />);
    expect(page.baseElement).toMatchSnapshot();
    expect(screen.getByText('Something went wrong')).toBeInTheDocument();
    expect(screen.getByText('Reload')).toBeInTheDocument();
  });
});
