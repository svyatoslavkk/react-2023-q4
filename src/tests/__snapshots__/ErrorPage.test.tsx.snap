// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for the 404 Page component > 404 page is displayed when navigating to an invalid route. 1`] = `
<body>
  <div>
    <div
      class="error-boundary-block"
    >
      <h2
        class="error-boundary-title"
      >
        Something went wrong
      </h2>
      <a
        href="/"
      >
        Reload
      </a>
    </div>
  </div>
</body>
`;
